src/Commands/core/LocaleCommands.php

```diff
@@ -208,6 +209,78 @@ public function exportValidate(CommandData $commandData): void
         }
     }
 
+    /**
+     * Imports multiple translation files from the defined directory.
+     *
+     * @throws \Exception
+     */
+    #[CLI\Command(name: self::IMPORT_ALL, aliases: ['locale-import-all', 'locale:import:all'])]
+    #[CLI\Argument(name: 'directory', description: 'The path to directory with translation files to import.')]
+    #[CLI\Option(name: 'type', description: 'String types to include, defaults to <info>not-customized</info>. Recognized values: <info>not-customized</info>, <info>customized</info>')]
+    #[CLI\Option(name: 'override', description: 'Whether and how imported strings will override existing translations. Defaults to the Import behavior configured in the admin interface. Recognized values: <info>none</info>, <info>customized</info>, <info>not-customized</info>, <info>all</info>')]
+    #[CLI\Usage(name: 'drush locale:import-all /var/www/translations', description: 'Import all translations from the defined directory (non-recursively). Supported filename patterns are: {project}-{version}.{langcode}.po, {prefix}.{langcode}.po or {langcode}.po.')]
+    #[CLI\Usage(name: 'drush locale:import-all /var/www/translations/custom --types=customized --override=all', description: 'Import all custom translations from the defined directory (non-recursively) and override any existing translation. Supported filename patterns are: {project}-{version}.{langcode}.po, {prefix}.{langcode}.po or {langcode}.po.')]
+    #[CLI\ValidateModulesEnabled(modules: ['locale'])]
+    public function importAll($directory, $options = ['type' => self::OPT, 'override' => self::OPT])
+    {
+        if (!is_dir($directory)) {
+            throw new \Exception('The defined directory does not exist.');
+        }
+
+        // Look for .po files in defined directory
+        $poFiles = glob($directory . DIRECTORY_SEPARATOR . '*.po');
+        if (empty($poFiles)) {
+            throw new \Exception('Translation files not found in the defined directory.');
+        }
+
+        $this->getModuleHandler()->loadInclude('locale', 'translation.inc');
+        $this->getModuleHandler()->loadInclude('locale', 'bulk.inc');
+
+        $translationOptions = _locale_translation_default_update_options();
+        $translationOptions['customized'] = $this->convertCustomizedType($options['type']);
+        $override = $this->convertOverrideOption($options['override']);
+        if ($override) {
+            $translationOptions['overwrite_options'] = $override;
+        }
+        $langcodes_to_import = [];
+        $files = [];
+        foreach ($poFiles as $file) {
+            // Ensure we have the file intended for upload.
+            if (file_exists($file)) {
```

Comment:
The other if conditions in this loop use `continue` to skip processing the current `$file`. I agree with using this style of code (early return). I'd like to use it consistently and apply it to this condition too, and remove these nested if conditions. With this being done, the comment 'Ensure we have the file ...' can go away. It is pretty obvious what is going on here.

Reply:
fixed